doc/psud/PSU_daemon_design.md

```diff
@@ -37,40 +49,103 @@ PSU daemon data collection flow diagram:
-    voltage_min_threshold   = 1*3.3DIGIT                     ; the minimal voltage threshold of the PSU
-    voltage_max_threshold   = 1*3.3DIGIT                     ; the maximum voltage threshold of the PSU
-    current                 = 1*3.3DIGIT                     ; the current of the PSU
-    power                   = 1*3.3DIGIT                     ; the power of the PSU
-    input_voltage           = 1*3.3DIGIT                     ; input voltage of the psu
-    input_current           = 1*3.3DIGIT                     ; input current of the psu
-    max_power               = 1*4.3DIGIT                     ; power capacity of the psu
-    
+    ; Defines information for a psu
+    key                       = PSU_INFO|psu_name              ; information for the psu
+    ; field                   = value
+    presence                  = BOOLEAN                        ; presence state of the psu
+    model                     = STRING                         ; model name of the psu
+    serial                    = STRING                         ; serial number of the psu
+    revision                  = STRING                         ; hardware revision of the PSU
+    status                    = BOOLEAN                        ; status of the psu
+    change_event              = STRING                         ; change event of the psu
+    fan                       = STRING                         ; fan_name of the psu
+    led_status                = STRING                         ; led status of the psu
+    is_replaceable            = STRING                         ; whether the PSU is replaceable
+    temp                      = 1*3.3DIGIT                     ; temperature of the PSU
+    temp_threshold            = 1*3.3DIGIT                     ; temperature threshold of the PSU
+    voltage                   = 1*3.3DIGIT                     ; the output voltage of the PSU
+    voltage_min_threshold     = 1*3.3DIGIT                     ; the minimal voltage threshold of the PSU
+    voltage_max_threshold     = 1*3.3DIGIT                     ; the maximum voltage threshold of the PSU
+    current                   = 1*3.3DIGIT                     ; the current of the PSU
+    power                     = 1*4.3DIGIT                     ; the power of the PSU
+    input_voltage             = 1*3.3DIGIT                     ; input voltage of the psu
+    input_current             = 1*3.3DIGIT                     ; input current of the psu
+    max_power                 = 1*4.3DIGIT                     ; power capacity of the psu
+    power_overload            = "true" / "false"               ; whether the PSU's power exceeds the threshold
+    power_warning_suppress_threshold   = 1*4.3DIGIT                     ; The power warning threshold
+    power_critical_threshold  = 1*4.3DIGIT                     ; The power critical threshold
 
 Now psud only collect and update "presence" and "status" field.
```

Comment:
we need one field in the DB which indicates the critical threshold warning is ACTIVE. this is needed for telemetry

Reply:
power_overload is the field to indicate whether it’s in the warning state